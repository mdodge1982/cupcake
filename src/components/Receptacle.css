.Receptacle {
	position: absolute;
	text-align: center;
	font-size: 20px;
	font-weight: bold;
	background-position: left top;
	background-repeat: no-repeat;
}
.trash {
	padding: 30px 3px 0 0;
	background-image: url('images/compost.png');
}
.box {
	padding: 20px 3px 0 0;
	background-image: url('images/box.png');
}
