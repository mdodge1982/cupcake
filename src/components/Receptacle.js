import React, {Component} from 'react';
import './Receptacle.css';

class Receptacle extends Component {
	render() {
		const {width,height,count,type,xPos,yPos,inline} = this.props;
		const style = {
			width: width+'px',
			height: height+'px'
		};
		if(inline){
			style.position = 'static';
			style.display = 'inline-block';
		}else{
			style.left = xPos+'px';
			style.bottom = yPos+'px';
		}
		return (
			<div className={'Receptacle '+type} style={style}>{count}</div>
		);
	}
};

export default Receptacle;
