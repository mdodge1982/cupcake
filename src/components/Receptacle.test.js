import React from 'react';
import renderer from 'react-test-renderer';
import Receptacle from './Receptacle';

const margin = 10;
const boxWidth = 67;

const trash = {
	width: 71,
	height: 100,
	xPos: margin,
	yPos: margin,
	type: 'trash',
	count: 5
};
const box = {
	width: boxWidth,
	height: 88,
	xPos: 525-boxWidth-margin,
	yPos: margin,
	type: 'box',
	count: 10
};
test('Initial render as trash', () => {
	const component = renderer.create(
		<Receptacle {...trash} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
test('Initial render as box', () => {
	const component = renderer.create(
		<Receptacle {...box} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Game over inline render', () => {
	const component = renderer.create(
		<Receptacle {...box} inline={true} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
