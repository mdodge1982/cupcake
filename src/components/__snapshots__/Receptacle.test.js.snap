// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game over inline render 1`] = `
<div
  className="Receptacle box"
  style={
    Object {
      "display": "inline-block",
      "height": "88px",
      "position": "static",
      "width": "67px",
    }
  }
>
  10
</div>
`;

exports[`Initial render as box 1`] = `
<div
  className="Receptacle box"
  style={
    Object {
      "bottom": "10px",
      "height": "88px",
      "left": "448px",
      "width": "67px",
    }
  }
>
  10
</div>
`;

exports[`Initial render as trash 1`] = `
<div
  className="Receptacle trash"
  style={
    Object {
      "bottom": "10px",
      "height": "100px",
      "left": "10px",
      "width": "71px",
    }
  }
>
  5
</div>
`;
